tools/analysis.py

```diff
@@ -22,77 +22,85 @@
+    end: Optional[int] = typer.Argument(None, help="End timestamp [epoch]"),
+    show: bool = typer.Option(True, help="Show plot rather than saving it"),
+    pdb: bool = typer.Option(False, help="Enter pdb debugger"),
+    params: bool = typer.Option(False, help="Use ./params.json file"),
+    gbg: bool = typer.Option(True, help="Set timezone to Gothenburg"),
+    sec: bool = typer.Option(False, help="use seconds rather than ms as time unit"),
 ):
     """Run an analyis on Source ID"""
-    
-    
+
     ses = MVG(ENDPOINT, TOKEN)
 
     # Fetch parameters if given
     if params:
         with open("params.json", "r", encoding="utf-8") as json_data:
             params = json.load(json_data)
-            print(f"Params loaded from ./params.json")
+            print("Params loaded from ./params.json")
             print(json.dumps(params, indent=4))
     else:
         params = {}
 
-    req = ses.request_analysis(sid=source_id,
-                               feature=feature,
-                               parameters=params,
-                               start_timestamp=start,
-                               end_timestamp=end)
+    req = ses.request_analysis(
+        sid=source_id,
+        feature=feature,
+        parameters=params,
+        start_timestamp=start,
+        end_timestamp=end,
+    )
     request_id = req["request_id"]
     print(f"Waiting for {request_id}")
     ses.wait_for_analyses([request_id])
-    get_and_display_results(ses, request_id, show, pdb)
+    get_and_display_results(ses, request_id, show, pdb, gbg, sec)
+
 
 @app.command()
 def retrieve(
-        request_id: str = typer.Argument(
-            ...,
-        help='Request ID'),
-        show: bool = typer.Option(True, help='Show plot rather than saving it'),
-        pdb: bool = typer.Option(False, help='Enter pdb debugger')
+    request_id: str = typer.Argument(..., help="Request ID"),
+    show: bool = typer.Option(True, help="Show plot rather than saving it"),
+    pdb: bool = typer.Option(False, help="Enter pdb debugger"),
+    gbg: bool = typer.Option(True, help="Set timezone to Gothenburg"),
+    sec: bool = typer.Option(False, help="use seconds rather than ms as time unit"),
 ):
     """Retrieve a previous analyis with request ID"""
-        
+
     ses = MVG(ENDPOINT, TOKEN)
-    get_and_display_results(ses, request_id, show, pdb)
+    get_and_display_results(ses, request_id, show, pdb, gbg, sec)
+
 
-    
-    
-def get_and_display_results(ses, request_id, show, pdb):
+def get_and_display_results(ses, request_id, show, pdb, gbg, sec):
 
     # Get results
     res_dict = ses.get_analysis_results(request_id)
 
+    # Time Unit
+    t_unit = "ms"
+    if sec:
+        t_unit = "s"
+
     # Parse results
-    res = parse_results(res_dict, "Europe/Stockholm", "s")
+    if gbg:
+        res = parse_results(res_dict, "Europe/Stockholm", t_unit)
+    else:
+        res = parse_results(res_dict, None, None)
```

Comment:
why `t_unit` ignored when timezone is not GBG?

Reply:
When --no-gbg indicates that no time zone information is available so no datetime conversion will be made. I used that for testing plots and data frames when no tz information is available.
The CLI tool only supported GBG (Europe/Stockholm) or no TZ.